internals/secrethub/migrate.go

```diff
@@ -450,38 +434,79 @@ func (cmd *MigrateApplyCommand) Run() error {
 				return err
 			}
 		} else {
-			if !cmd.append {
-				confirmed, err := ui.AskYesNo(cmd.io, fmt.Sprintf("A vault named %s already exists. Do you want to append the items to that vault?", vault.Name), ui.DefaultYes)
-				if err != nil {
-					return fmt.Errorf("a vault named %s already exists. Use --apend to append to existing vaults", vault.Name)
-				}
-				if !confirmed {
-					fmt.Fprintln(cmd.io.Output(), "Aborting.")
-					return nil
-				}
-			}
```

Comment:
That's right, I totally forgot to account for existing vaults.

I removed the check because:
1. When you run the migration tool a second time it will prompt you for each vault that it created when you ran it the first time. This prompting is not necessary as we already prompt the user for each field/secret that we update.
2. We cannot determine if this is the first time the migration tool is being run or if it has been run before, so asking if the tool is allowed to "append to and existing vault" does not make sense anymore since it could either be appending new items (if this is the first run of the migration tool) or updating existing ones (on subsequent runs).

My suggestion would be to only ask the user if it is ok to append to an existing vault if we actually find an item or field that is missing and we want to append it. 

This would mean that when you first run the migration tool, if you have vaults with the same name as your SecretHub repos you will be asked, just like before, if you want to append to the existing vaults. In the rare situation in which your vault, item and field names match with the ones generated by the migration tool you will still be asked if you want to update these values to their SecretHub counterparts.

In subsequent runs, most of the time no new items will be appended so you will not be bothered by the constant prompting.

What do you think?

Reply:
We discussed this on a call and came up with:
1. Plan all changes to create/append/update before executing any operation.
2. Prompt for confirmation if there's going to be any appending/updating. We'll have one prompt for all changes to be made.
3. Add an `--update` flag which skips the prompt, to be used in headless mode, e.g. to continuously sync the values of fields on SecretHub with 1Password using a cronjob.

Reply:
Yes, thanks for the summary!

Note, however that this could fail if the state of SecretHub or 1Password changes after the "plan" but before applying the changes. I don't think it's a huge issue, just pointing it out as you might think otherwise.

Reply:
There's only a use case for the migration commands if SecretHub is still the source of truth for these secrets, so we can assume there's no updates/writes on the 1Password side to the secrets we're migrating.

Also, I'm okay with assuming no two people/machines execute the `apply` command simultaneously.

So we don't require locking of some sort to prevent any writes on the 1Password side in between the plan and apply of the apply command (we need a better name for this "plan" as it does not refer to the plan command).

Reply:
This is currently the output when creating new vaults:
```
[2/2] Checking vault: my-project-prd

Vault my-project-dev:
  Create vault 'my-project-dev'
  Create item 'stripe-api-token'
  Create item 'aws'
  Create item 'db'
Vault my-project-prd:
  Create vault 'my-project-prd'
  Create item 'stripe-api-token'
  Create item 'db'
  Create item 'aws'

12 fields will be created
0 fields will be updated
0 fields are already up-to-date
0 fields will be skipped

Would you like to apply these changes? [y/N]: y
```

And this is the output when appending to existing vaults:
```
[2/2] Checking vault: my-project-prd

Vault my-project-dev:
  Create item 'stripe-api-token'
  Create item 'aws'
  Create item 'db'
Vault my-project-prd:
  Create item 'stripe-api-token'
  Create item 'db'
  Create item 'aws'

12 fields will be created
0 fields will be updated
0 fields are already up-to-date
0 fields will be skipped

Would you like to apply these changes? [y/N]:
```

Can we make it extra clear we're appending to an existing vault? I believe the omission of a "Create vault" in this list is not enough to make the user aware of what is happening. 

Remember the two mentioned above:
1. SecretHub customers that are already 1Password customers before the acquisition already have existing vaults. We don't want to mix SecretHub secrets with the secrets already present in these existing vaults without the users explicit consent.
2. The existing vault can already be shared with other people/machines. We don't want to share the secrets that are being imported without the users consent.

I believe we can improve this by calling out more explicitly that we're appending to an existing vault.

Two more things we may consider doing:
1. Report if there's items in the vault that aren't in the plan
2. Report if the vault is already shared with more members on the team

I consider reporting that we're appending blocking to get this merged. Reporting other items present and the vault already being shared is not blocking imo, we can evaluate to add that separately if we want to.

Reply:
How about something like this?
```
Vault my-project-dev (new vault):
  Create vault 'my-project-dev'
  Create item 'stripe-api-token'
  Create item 'aws'
  Create item 'db'
Vault my-project-prd (already exists):
  Create item 'stripe-api-token'
  Create item 'db'
  Create item 'aws'

Warning the following secrets will be added to a shared vault and will be accessible to other users:
  my-project-prd.stripe-api-token
  my-project-prd.db
  my-project-prd.aws
```

I am a bit hesitant to say `Create item` when the item is in a new vault and `Append item` when the vault already exists because there is no difference between the two operations so this might cause confusion.

Reply:
It is difficult to determine which vaults are accessible directly or indirectly by other users so I just added a note.